File: solution_01.rb
# frozen_string_literal: true

INPUT = File.read('input_01.txt')

module Solution
  extend self
  def groups(input = INPUT)
    input.split("\n\n").map { |group| group.split.sum(&:to_i) }
  end

  def part1
    groups.max
  end

  def part2
    groups.sort.max(3).sum
  end
end

puts "Part 1: #{Solution.part1}"

puts "Part 2: #{Solution.part2}"

File: solution_01.exs
defmodule Solution do
  @input File.read!("input_01.txt") |> String.trim()

  def input do
    @input
    |> String.split("\n\n")
    |> Enum.map(fn e -> String.split(e, "\n") |> Enum.map(&String.to_integer/1) |> Enum.sum end)
  end

  def solve_01 do
    input() |> Enum.max()
  end

  def solve_02 do
    input()
    |> Enum.sort()
    |> Enum.reverse()
    |> Enum.take(3)
    |> Enum.sum()
  end
end

IO.puts "First part: #{Solution.solve_01()}"
IO.puts "Second part: #{Solution.solve_02()}"
